gmso/lib/potential_templates.py

```diff
@@ -1,158 +1,130 @@
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
-
-class PeriodicTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='PeriodicTorsionPotential',
-                 expression='k * (1 + cos(n * phi - phi_eq))**2',
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
-
-class OPLSTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='OPLSTorsionPotential',
-                 expression=('0.5 * k0 + '
-                             '0.5 * k1 * (1 + cos(phi)) + '
-                             '0.5 * k2 * (1 - cos(2*phi)) + '
-                             '0.5 * k3 * (1 + cos(3*phi)) + '
-                             '0.5 * k4 * (1 - cos(4*phi))'),
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
-
-class RyckaertBellemansTorsionPotential(PotentialTemplate):
-    def __init__(self,
-                 name='RyckaertBellemansTorsionPotential',
-                 expression=('c0 * cos(phi)**0 + c1 * cos(phi)**1 + '
-                             'c2 * cos(phi)**2 + c3 * cos(phi)**3 + '
-                             'c4 * cos(phi)**4 + c5 * cos(phi)**5'),
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
+            template=template,
         )
 
-class HarmonicImproperPotential(PotentialTemplate):
-    def __init__(self,
-                 name='HarmonicImproperPotetial',
-                 expression='0.5 * k * (phi - phi_eq)**2',
-                 independent_variables={'phi'}):
-
-        super(PotentialTemplate, self).__init__(
-                name=name,
-                expression=expression,
-                independent_variables=independent_variables,
-                template=True
-        )
 
+class PotentialTemplates(Singleton):
+    """A singleton collection of all the potential templates"""
+
+    def __init__(self):
+        try:
+            self.json_refs
+        except AttributeError:
+            self.json_refs = POTENTIAL_JSONS
+            self.user_jsons = []
+            potential_names = [(os.path.split(filename)[-1]).replace('.json', '') for filename in POTENTIAL_JSONS]
```

Comment:
I think this is a case in which `pathlib` can be a little more elegant than chopping off the extension and splitting the string up

Reply:
Agreed.